finality-aleph/src/tcp_network.rs

```diff
@@ -0,0 +1,127 @@
+use std::{io::Result as IoResult, net::ToSocketAddrs as _};
+
+use aleph_primitives::AuthorityId;
+use codec::{Decode, Encode};
+use log::info;
+use tokio::net::{
+    tcp::{OwnedReadHalf, OwnedWriteHalf},
+    TcpListener, TcpStream, ToSocketAddrs,
+};
+
+use crate::{
+    network::{Multiaddress, NetworkIdentity, PeerId},
+    validator_network::{Dialer, Listener, Splittable},
+};
+
+impl Splittable for TcpStream {
+    type Sender = OwnedWriteHalf;
+    type Receiver = OwnedReadHalf;
+
+    fn split(self) -> (Self::Sender, Self::Receiver) {
+        let out = self.into_split();
+        (out.1, out.0)
```

Comment:
May we use here something different than bare pair syntax?

Reply:
Done.